patroni/__init__.py

```diff
@@ -48,13 +48,19 @@ def load_dynamic_configuration(self):
                 logger.warning('Can not get cluster from dcs')
 
     def get_tags(self):
-        return {tag: value for tag, value in self.config.get('tags', {}).items()
+        tags = {tag: value for tag, value in self.config.get('tags', {}).items()
                 if tag not in ('clonefrom', 'nofailover', 'noloadbalance') or value}
+        tags.update(self.ha.get_dynamic_tags())
```

Comment:
`get_tags` method executed only on patroni start and "reload". Result returned by this method is assigned to the `Patroni.tags` and is used in different places (API and touch_member).
But none of them will see nosync tag if reload didn't happened.